Makefile

```diff
@@ -6,3 +7,19 @@ fmt:
 
 deploy: config
 	firebase deploy
+
+# --- Install software ----
+
+software-update: firebase-update jsonnet-update  ## Install/update tools needed by this Makefile
+
+fbos_Linux = linux
+fbos_Dawrin = macos
+fbinstall = /usr/local/bin/firebase
+firebase-update:  ## Install/update firebase CLI to $(fbinstall)
+	curl -L -o '$(fbinstall)' 'https://firebase.tools/bin/$(fbos_$(shell uname))/latest'
```

Comment:
can you not do some `lower(uname)` magic rather than defining these extra vars above?

Reply:
no because lower("Darwin") != "macos". I essentially wanted a map and that's how you do a map in make. happy to entertain other ideas, but rather not spend too much time on it. this is not a repo that will see a lot of use i imagine.

Reply:
doh, sorry! lgtm.